core/src/value/fragments.rs

```diff
@@ -175,4 +179,32 @@ mod tests {
         assert_eq!(fragment.fragments[1].len(), 2);
         assert_eq!(fragment.fragments[1], vec![200, 222]);
     }
+
+    #[test]
+    fn test_bot_single_fragment_generation() {
+        let data = vec![Fragments::new(vec![0u8, 0u8], 2)];
+        let fragment_sequence: PixelFragmentSequence<InMemFragment> = data.into();
+        assert_eq!(fragment_sequence.offset_table.len(), 1);
+        assert_eq!(fragment_sequence.offset_table[0], 0);
+    }
+
+    #[test]
+    fn test_bot_multi_fragments_generation() {
+        let data = vec![Fragments::new(vec![0u8, 0u8, 0u8, 0u8], 2)];
+        let fragment_sequence: PixelFragmentSequence<InMemFragment> = data.into();
+        assert_eq!(fragment_sequence.offset_table.len(), 1);
+        assert_eq!(fragment_sequence.offset_table[0], 0);
+    }
+
+    #[test]
+    fn test_bot_multi_frame_generation() {
+        let data = vec![
+            Fragments::new(vec![0u8, 0u8, 0u8, 0u8], 0),
+            Fragments::new(vec![1u8, 1u8, 1u8, 1u8, 1u8, 1u8], 0),
```

Comment:
This might be easier to read. The other `vec!` calls may be refactored as well.

```suggestion
            Fragments::new(vec![0u8; 4], 0),
            Fragments::new(vec![1u8; 6], 0),
```